### PR TITLE
[MQ] Add overload for unimplemented to suport async functions

### PR DESCRIPTION
**Pull request by ggumieniakMQ**
Create overloaded function in Unimplemented.swift for async, throwing and async throwing situations
Change word in StaticString+IsEmpty as more describing value

### CODE REVIEW

File: Sources/MQ/Errors/Unimplemented.swift [hunk 1]
```diff
@@ -58,6 +58,89 @@ public func unimplemented(
...
+/// - Parameters:
+///   - message: Message associated with this error.
+///   Default value is "Unimplemented".
+///   - file: Source code file identifier.
+///   Filled automatically based on compile time constants.
+///   - line: Line in given source code file.
+///   Filled automatically based on compile time constants.
+/// - Throws: Unimplemented error.
+/// - Returns: Function throwing always the same ``Unimplemented`` error when executed.
+public func unimplementedAsyncThrowing(
```
Comment (KaQuMiQ) [inline Sources/MQ/Errors/Unimplemented.swift:130]: and we also don't need this one as well (similarly to the previous one)

File: Sources/MQ/Errors/Unimplemented.swift [hunk 2]
```diff
@@ -118,8 +201,8 @@ public func unimplemented<A1, R>(
 
 /// Convenient placeholder for unimplemented function.
 ///
-/// Placeholder for a function indicating unimplemented part of code.
-/// Terminates process with ``Unimplemented`` error as the cause when the function becomes executed.
+/// Placeholder for a function indicating unimplemented throwing part of code.
+/// Throw ``Unimplemented`` error as the cause.
```
Comment (KaQuMiQ) [inline Sources/MQ/Errors/Unimplemented.swift:205]: Please replace "Throw ``Unimplemented`` error as the cause." with "Throws ``Unimplemented`` error when executed." (applies to all throwing overloads)

**Review by KaQuMiQ — CHANGES_REQUESTED** [review]

**Review by ggumieniakMQ — COMMENTED** [review]
Changes was added, please once again check if now it's everything fine.

File: Sources/MQ/Errors/Unimplemented.swift [hunk 3]
```diff
@@ -87,6 +87,93 @@ public func unimplemented<R>(
...
+				line: line
+			)
+			.asFatalError()
+	}
+}
+
+/// Convenient placeholder for unimplemented function.
+///
+/// Placeholder for a function indicating unimplemented throwing part of code.
+/// Throws ``Unimplemented`` when executed.
```
Comment (KaQuMiQ) [inline Sources/MQ/Errors/Unimplemented.swift:122]: Please change a little bit more from "Placeholder for a function indicating unimplemented throwing part of code." to "Placeholder for a function indicating unimplemented part of code.".
And for next line please change from "Throws ``Unimplemented`` when executed." to "Throws ``Unimplemented`` when the function becomes executed."
It applies for all throwing functions.

File: Sources/MQ/Errors/Unimplemented.swift [hunk 4]
```diff
@@ -87,6 +87,93 @@ public func unimplemented<R>(
...
+///
+/// - Parameters:
+///   - message: Message associated with this error.
+///   Default value is "Unimplemented".
+///   - file: Source code file identifier.
+///   Filled automatically based on compile time constants.
+///   - line: Line in given source code file.
+///   Filled automatically based on compile time constants.
+/// - Returns: Function throwing ``Unimplemented`` error when executed.
+public func unimplementedAsyncThrow<R>(
```
Comment (KaQuMiQ) [inline Sources/MQ/Errors/Unimplemented.swift:161]: please rename `unimplementedAsyncThrow` -> `unimplementedAsyncThrowing` - applies to all async throwing

**Review by KaQuMiQ — CHANGES_REQUESTED** [review]

File: Sources/MQ/Errors/Unimplemented.swift (hunk 4 above)
Comment (ggumieniakMQ) [reply Sources/MQ/Errors/Unimplemented.swift]: There were only one functions in async throwing which was labeled like that. Change was made.

File: Sources/MQ/Errors/Unimplemented.swift (hunk 3 above)
Comment (ggumieniakMQ) [reply Sources/MQ/Errors/Unimplemented.swift:122]: OK

**Review by KaQuMiQ — APPROVED** [review]

**Pull request merged**